src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -12,13 +69,494 @@ import { ActivatedRoute } from '@angular/router';
+      this.toggleTeamState(team, state);
+    });
+  }
+
+  /**
+   * Function to change the state of a specific team.
+   */
+  toggleTeamState(team: TeamDetails, designatedState?: boolean): void {
+    const state: boolean = designatedState === undefined
+        ? !team.isChecked
+        : designatedState;
+
+    team.isChecked = state;
+
+    if (!this.isTeamDetailsInList(team)) {
+      this.teamDetailsList.push(team);
+    }
+
+    team.students.forEach((student: StudentDetails) => {
+      this.toggleStudentState(student, state);
+    });
+  }
+
+  /**
+   * Function to change the state of a specific student.
+   */
+  toggleStudentState(student: StudentDetails, designatedState?: boolean): void {
+    const state: boolean = designatedState === undefined
+        ? !student.isChecked
+        : designatedState;
+
+    student.isChecked = state;
+
+    if (!this.isStudentDetailsInList(student)) {
+      this.studentDetailsList.push(student);
+    }
+  }
+
+  /*------------------------------Functions to get item state from state maps-----------------------------------*/
+  /**
+   * Function to get the list of students which need to be hidden on the StudentTable display.
+   */
+  getStudentsToHide(courseId: string): string[] {
+    const studentsToHide: string[] = this.studentDetailsList
+        .filter((student: StudentDetails) => student.courseId === courseId && !student.isChecked)
+        .map((student: StudentDetails) => student.email);
+
+    return studentsToHide;
+  }
+
+  /*------------------------------Functions to formulate data for student list from item------------------------*/
+  /**
+   * Function to formulate data for student list.
+   */
+  getStudentListSectionDataForCourse(courseId: string): StudentListSectionData[] {
+    const data: StudentListSectionData[] | undefined = this.courseStudentListSectionDataMap[courseId];
+    if (data === undefined) {
+      return [];
+    }
+
+    return data;
+  }
+
+  /**
+   * Function to formulate data for student list.
+   */
+  getStudentListSectionDataFromCourseDetails(courseDetails: CourseDetails): StudentListSectionData[] {
+    const sections: StudentListSectionData[] = [];
+    courseDetails.sections.forEach((section: SectionDetails) => {
+      const sectionData: StudentListSectionData = this.mapSectionForStudentList(section);
+      sections.push(sectionData);
+    });
+
+    return sections;
+  }
+
+  /**
+   * Function to formulate data for student list from SectionDetails.
+   */
+  mapSectionForStudentList(sectionDetails: SectionDetails): StudentListSectionData {
+    const students: StudentListStudentData[] = [];
+
+    sectionDetails.teams.forEach((team: TeamDetails) => {
+      team.students.forEach((student: StudentDetails) => {
+        const studentData: StudentListStudentData = this.mapStudentForStudentList(student);
+        students.push(studentData);
+      });
+    });
+
+    const section: StudentListSectionData = {
```

Comment:
You could just return this directly. If the typing gives you problem, you should mention `as StudentListSectionData`.